### ISSUE TITLE
/validate/triggerchallenge fails for WebAuthn tokens

### ISSUE THREAD

**Issue by NuvandaPV** [privacyidea/privacyidea#2329]
`triggerchallenge()` fails to populate `options` with all options necessary to create a challenge for a WebAuthn token.

**Comment by NuvandaPV** [comment]
`check()` populates the `options` with values from `request.all_data` [here](https://github.com/privacyidea/privacyidea/blob/42e673c9cc4a8422db5be164dec3c2b12759ab30/privacyidea/api/validate.py#L368). Adding the equivalent code [here](https://github.com/privacyidea/privacyidea/blob/42e673c9cc4a8422db5be164dec3c2b12759ab30/privacyidea/api/validate.py#L506) solves the initial problem.

However, the code runs into errors, when trying to `get_tokeninfo()` later:

```
[2020-08-18 14:50:03,755][6030][140085391951680][ERROR][privacyidea.app:1891] Exception on /validate/triggerchallenge [POST]
Traceback (most recent call last):
  File "/home/user/Desktop/privacyidea/venv/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/user/Desktop/privacyidea/venv/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/user/Desktop/privacyidea/venv/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/user/Desktop/privacyidea/venv/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/user/Desktop/privacyidea/venv/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/user/Desktop/privacyidea/venv/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/user/Desktop/privacyidea/privacyidea/api/auth.py", line 366, in decorated_function
    return f(*args, **kwargs)
  File "/home/user/Desktop/privacyidea/privacyidea/api/lib/postpolicy.py", line 107, in policy_wrapper
    response = wrapped_function(*args, **kwds)
  File "/home/user/Desktop/privacyidea/privacyidea/api/lib/postpolicy.py", line 107, in policy_wrapper
    response = wrapped_function(*args, **kwds)
  File "/home/user/Desktop/privacyidea/privacyidea/lib/decorators.py", line 100, in check_user_or_serial_in_request_wrapper
    f_result = func(*args, **kwds)
  File "/home/user/Desktop/privacyidea/privacyidea/api/lib/prepolicy.py", line 154, in policy_wrapper
    return wrapped_function(*args, **kwds)
  File "/home/user/Desktop/privacyidea/privacyidea/api/lib/prepolicy.py", line 154, in policy_wrapper
    return wrapped_function(*args, **kwds)
  File "/home/user/Desktop/privacyidea/privacyidea/api/lib/prepolicy.py", line 154, in policy_wrapper
    return wrapped_function(*args, **kwds)
  [Previous line repeated 1 more time]
  File "/home/user/Desktop/privacyidea/privacyidea/lib/event.py", line 99, in event_wrapper
    f_result = func(*args, **kwds)
  File "/home/user/Desktop/privacyidea/privacyidea/api/validate.py", line 513, in trigger_challenge
    create_challenges_from_tokens(chal_resp_tokens, details, options)
  File "/home/user/Desktop/privacyidea/privacyidea/lib/token.py", line 2046, in create_challenges_from_tokens
    transactionid=transaction_id, options=options)
  File "/home/user/Desktop/privacyidea/privacyidea/lib/tokens/webauthntoken.py", line 1064, in create_challenge
    user = self._get_webauthn_user(getParam(options, "user", required))
  File "/home/user/Desktop/privacyidea/privacyidea/lib/tokens/webauthntoken.py", line 761, in _get_webauthn_user
    public_key=webauthn_b64_encode(binascii.unhexlify(self.get_tokeninfo(WEBAUTHNINFO.PUB_KEY))),
TypeError: argument should be bytes, buffer or ASCII string, not 'NoneType'
```

**Comment by plettich** [comment]
i could not reproduce this after adding `all_data` to `options` and with a completely enrolled webauchn-token.

**Comment by plettich** [comment]
duplicate of #2538 and fixed in #2547
